src/server/pfs/server/paths.go

```diff
@@ -0,0 +1,71 @@
+package server
+
+import (
+	"path"
+	"regexp"
+	"strings"
+
+	globlib "github.com/pachyderm/ohmyglob"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset/index"
+)
+
+var globRegex = regexp.MustCompile(`[*?[\]{}!()@+^]`)
+
+func globLiteralPrefix(glob string) string {
+	idx := globRegex.FindStringIndex(glob)
+	if idx == nil {
+		return glob
+	}
+	return glob[:idx[0]]
+}
+
+func matchFunc(glob string) (func(string) bool, error) {
+	// TODO Need to do a review of the globbing functionality.
+	var parentG *globlib.Glob
+	parentGlob, baseGlob := path.Split(glob)
+	if len(baseGlob) > 0 {
+		var err error
+		parentG, err = globlib.Compile(parentGlob, '/')
+		if err != nil {
+			return nil, err
+		}
+	}
+	g, err := globlib.Compile(glob, '/')
+	if err != nil {
+		return nil, err
+	}
+	return func(s string) bool {
+		s = path.Clean(s)
```

Comment:
Why is this here?

Reply:
I think that was to get the match function working on the tar names before path cleaning. In the latest commit the match function is only used in GetTarConditional.

Reply:
Ohh, I see. We will need to unify this functinality, but not in this PR.